core/src/data/interval/mod.rs

```diff
@@ -287,7 +287,7 @@ impl Interval {
                 format!("{:?}", to),
             )
             .into()),
-            (None, _) => Err(IntervalError::Unreachable.into()),
+            (_, _) => Err(IntervalError::UnsupportedExtractPart.into()),
```

Comment:
Is this code reachable? Then, we need to have a test case for generting this `IntervalError::UnsupportedExtractPart`